inst/extdata/installPackages.R

```diff
@@ -0,0 +1,129 @@
+# Install repgen, production instance, prerequisite packages. Should be safe to
+# run repeatedly without additional intervention.
+
+# set to TRUE if you intend to develop repgen on this system
+developer <- FALSE
+
+# devtools is currently only required on a production instance to install gsplot
+# from GitHub (see below). There are plans to "promote" it to CRAN (or GRAN?),
+# but that hasn't happened yet. See 
+# https://usgs-cida.slack.com/archives/aqcu/p1484934743005621 if it still 
+# exists.
+pkgs <- c(
+  "devtools",
+  "dplyr",
+  "evaluate",
+  "htmlTable",
+  "jsonlite",
+  "lazyeval",
+  "knitr",
+  "lubridate",
+  "rmarkdown",
+  "yaml"
+)
+
+# convenience wrapper function around install.packages()
+installPackages <- function(pkgs, lib, repos = getOption("repos")) {
+  for (p in pkgs) {
+    tryCatch({
+      packageDescription(p) # check to see if package p is already installed
+    },
+    warning = function(w) {
+      # if package p is not installed...
+      if (any(grepl("(DESCRIPTION file of package .+ is missing or broken|no package .+ was found)", w))) {
+        # ...install it
+        install.packages(p, lib, repos = repos)
+      } else {
+        print(w)
+      }
+    },
+    error = function(e) {
+      print(e)
+    })
+  }
+}
+
+# if this is a production tier...
+if (!developer) {
+  lib <- Sys.getenv("R_LIBS")
+  if (nchar(lib) == 0) {
+    stop(
+      paste(
+        "Could not get a value for R_LIBS environment variable; is this a",
+        "development system?",
+        sep = "\n  "
+      )
+    )
+  }
+} else {
+  libPaths <- .libPaths()
+  lib <- libPaths[1]
+}
+
+# all packages except devtools and its prerequisites are held back to older
+# versions, keyed by their repo URLs below
+installPackages(pkgs, lib, "http://mran.microsoft.com/snapshot/2016-03-31")
+installPackages("httr", lib, "http://mran.microsoft.com/snapshot/2016-01-27")
+
+# To be able to install gsplot from GitHub using devtools (below), download
+# DOI root certificate and append to openssl package's certificate bundle
+# (ruthlessly, without asking). See also
+# https://usgs-cida.slack.com/archives/r-activities/p1469472383000332
+
+# locate & read the openssl certificate bundle
+cert_bundle <- httr:::find_cert_bundle()
```

Comment:
this might be unnecessary on the server but I am not entirely sure. the doi cert _should_ already be there, and I doubt the R user or Tomcat has the power to edit it if it found it this way?